File: source/dotnet/Library/AdaptiveCards/Rendering/AdaptiveCardConfig.cs
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
using System.Text.Json.Serialization;

namespace AdaptiveCards.Rendering
{
    /// <summary>
    /// Contains options for the AdaptiveCard element.
    /// </summary>
    public class AdaptiveCardConfig : AdaptiveConfigBase
    {
        /// <summary>
        /// Determines whether custom styles should be honored.
        /// </summary>
        [JsonIgnore(Condition = JsonIgnoreCondition.WhenWritingDefault)]
        public bool AllowCustomStyle { get; set; }
    }
}

File: source/uwp/SharedRenderer/lib/AdaptiveCardConfig.cpp
// Copyright (c) Microsoft Corporation. All rights reserved.
// Licensed under the MIT License.
#include "pch.h"

#include "AdaptiveCardConfig.h"
#include "AdaptiveCardConfig.g.cpp"

namespace winrt::AdaptiveCards::Rendering::Xaml_Rendering::implementation
{
    AdaptiveCardConfig::AdaptiveCardConfig(::AdaptiveCards::AdaptiveCardConfig const& adaptiveCardConfig)
    {
        AllowCustomStyle = adaptiveCardConfig.allowCustomStyle;
    }
}
